frontend/src/pages/RegisterPage.tsx

```diff
@@ -0,0 +1,73 @@
+import {Box, Button, Paper, TextField} from "@mui/material"
+import {Typography, Alert} from "@mui/material"
+import {FormEvent, useState} from "react"
+import {useRegisterMutation} from "../features/api/apiSlice.tsx"
+import {useNavigate} from "react-router-dom"
+
+function RegisterPage() {
+    const [username, setUsername] = useState<string>("")
+    const [email, setEmail] = useState<string>("")
+    const [password, setPassword] = useState<string>("")
+    const [repeatPassword, setRepeatPassword] = useState<string>("")
+    const [register, {isError, isSuccess, data, error}] = useRegisterMutation()
```

Comment:
What does the apiSlice.tsx do?

Reply:
it handles the frontend request logic to the backend